src/Modal.js

```diff
@@ -55,7 +55,7 @@ const Modal = React.createClass({
         className={classNames(this.props.className, classes)}
         onClick={this.props.backdrop === true ? this.handleBackdropClick : null}
         ref="modal">
-        <div className={classNames(dialogClasses)}>
+        <div className={classNames(this.props.dialogClassName, dialogClasses)}>
```

Comment:
I have not seen such failures with `eslint-plugin-react`.

I don't see a problem with this code:
```jsx
propTypes: {
  ...
  dialogClassName: React.PropTypes.string // type check
}
... // and then it is used
<div className={classNames(this.props.dialogClassName, dialogClasses)}>
```

Or type checking wasn't there at the first place ?
Then I'll do more thorough investigation about the situation.